tests/expectations/metrics/test_metric_providers.py

```diff
@@ -0,0 +1,315 @@
+
+    CustomColumnValuesEqualSeven()
+
+    assert len(_registered_metrics.keys()) == prev_registered_metric_key_count + 8
+
+    new_keys = [
+        "column_values.equal_seven.condition",
+        "column_values.equal_seven.unexpected_count",
+        "column_values.equal_seven.unexpected_index_list",
+        "column_values.equal_seven.unexpected_index_query",
+        "column_values.equal_seven.unexpected_rows",
+        "column_values.equal_seven.unexpected_values",
+        "column_values.equal_seven.unexpected_value_counts",
+        "column_values.equal_seven.unexpected_count.aggregate_fn",
+    ]
+    for key in new_keys:
+        assert key in _registered_metrics.keys()
+
+
+def test__column_pair_map_metric__registration():
+    """This tests whether the ColumnPairMapMetricProvider class registers the correct metrics."""
+
+    registered_metric_keys = list(_registered_metrics.keys())
+    for key in registered_metric_keys:
+        assert "column_pair_values.equal_seven" not in key
+    prev_registered_metric_key_count = len(registered_metric_keys)
+
+    class CustomColumnPairValuesEqualSeven(ColumnPairMapMetricProvider):
+        condition_metric_name = "column_pair_values.equal_seven"
+
+        @column_pair_condition_partial(engine=PandasExecutionEngine)
+        def _pandas(cls, column_A, column_B, **kwargs):
+            raise NotImplementedError
+
+        @column_pair_condition_partial(engine=SqlAlchemyExecutionEngine)
+        def _sqlalchemy(cls, column_A, column_B, _dialect, **kwargs):
+            raise NotImplementedError
+
+        @column_pair_condition_partial(engine=SparkDFExecutionEngine)
+        def _spark(cls, column_A, column_B, **kwargs):
+            raise NotImplementedError
+
+    CustomColumnPairValuesEqualSeven()
+
+    assert len(_registered_metrics.keys()) == prev_registered_metric_key_count + 7
+
+    for key in _registered_metrics.keys():
+        if "column_pair_values.equal_seven" in key:
+            print(key)
+
+    new_keys = [
+        "column_pair_values.equal_seven.condition",
+        "column_pair_values.equal_seven.unexpected_count",
+        "column_pair_values.equal_seven.unexpected_index_list",
+        "column_pair_values.equal_seven.unexpected_index_query",
+        "column_pair_values.equal_seven.unexpected_rows",
+        "column_pair_values.equal_seven.unexpected_values",
+        "column_pair_values.equal_seven.filtered_row_count",
+    ]
+    for key in new_keys:
+        assert key in _registered_metrics.keys()
+
+
+def test__multicolumn_map_metric__registration():
+    """This tests whether the MultiColumnMapMetricProvider class registers the correct metrics."""
+
+    registered_metric_keys = list(_registered_metrics.keys())
+    for key in registered_metric_keys:
+        assert "multicolumn_values.equal_seven" not in key
+    prev_registered_metric_key_count = len(registered_metric_keys)
+
+    class CustomColumnPairValuesEqualSeven(MulticolumnMapMetricProvider):
```

Comment:
@abegong There appears to be a typo in the intended test class name here (for illustrative purposes, it was probably intended to be named something like `CustomMulticolumnValuesEqualSeven`.  Thank you.